internal/conduit/pipeline/oracle.go

```diff
@@ -58,14 +63,21 @@ func NewOracle(ctx context.Context, ot OracleType,
 	return o, nil
 }
 
+func (o *Oracle) Close() {
+	log.Printf("Waiting for oracle goroutines to be done.")
+	o.waitGroup.Wait()
+	log.Printf("Oracle goroutines have exited.")
+}
```

Comment:
Let's create an issue to fix this closure logic and add closures for all component types

Reply:
Created #22